Refactor `gulpfile.js`

Raruto opened the pull request:
a partial refactor is actually needed (or at least suggested) before complete [#68](https://github.com/g3w-suite/g3w-client/issues/68)

based on https://github.com/g3w-suite/g3w-client/pull/35

---

**Notable changes:**

- remove function `interpolateVersion()` 
- remove function `set_current_hash_version()`
- remove task `g3w-admin:client_only`
- remove task `set_build_all_to_false`
- remove variable `g3w_admin_version`

Raruto committed:
rename `conf` variable into `g3w` within gulpfile.js

Raruto committed:
reorder CommonJS requires within `gulpfile.js`

Raruto committed:
refactor gulpfile.js

a partial refactor is actually needed (or at least suggested) before complete https://github.com/g3w-suite/g3w-client/issues/68

based on https://github.com/g3w-suite/g3w-client/pull/35

Raruto committed:
Merge branch 'dev' into gulpfile-refactor

Raruto commented on gulpfile.js:
```diff
@@ -1,58 +1,70 @@
+/**
+ * TODO: can we safely delete the followings ?
+ */
+
```
@volterra79 can you check this before merge?

Raruto commented on gulpfile.js:
```diff
@@ -71,422 +83,496 @@ const buildChanges = {
+ * TODO: check which of the following tasks is actually required
+ */
+//////////////////////////////////////////////////////////////////////////
+
```
@volterra79 can you check this before merge?

volterra79 replied:
ok .. i let you know as soon as possible

volterra79 committed:
Merge branch 'dev' into gulpfile-refactor

# Conflicts:
#	gulpfile.js

volterra79 committed:
Remove unused function and variables.
Simplify tasks: we use only dev and build task without specify if we wold like to create app and vendors files.

volterra79 committed:
Merge branch 'dev' into gulpfile-refactor

volterra79 committed:
Remove build:client and admin:client script to simplify tasks and script

Raruto commented:
@volterra79 if that's okay with you I'd say we can quite safely merge this.
